Sources/Subprocess/Platforms/Subprocess+Windows.swift

```diff
@@ -629,6 +750,148 @@ extension Executable {
+    /// This method replicates the executable searching behavior of `CreateProcessW`'s
+    /// `lpCommandLine`. Specifically, it follows the steps listed in `CreateProcessW`'s documentation:
+    ///
+    /// 1. The directory from which the application loaded.
+    /// 2. The current directory for the parent process.
+    /// 3. The 32-bit Windows system directory.
+    /// 4. The 16-bit Windows system directory.
+    /// 5. The Windows directory.
+    /// 6. The directories that are listed in the PATH environment variable.
+    ///
+    /// For more info:
+    /// https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw
+    internal func possibleExecutablePaths(
+        withPathValue pathValue: String?
+    ) -> Set<String> {
+        func insertExecutableAddingExtension(
+            _ name: String,
+            currentPath: String,
+            pathExtensions: Set<String>,
+            storage: inout Set<String>
+        ) {
+            let fullPath = FilePath(currentPath).appending(name)
+            if !name.hasExtension() {
+                for ext in pathExtensions {
+                    var path = fullPath
+                    path.extension = ext
+                    storage.insert(path.string)
+                }
+            } else {
+                storage.insert(fullPath.string)
+            }
+        }
+
+        switch self.storage {
+        case .executable(let name):
+            var possiblePaths: Set<String> = []
+            let currentEnvironmentValues = Environment.currentEnvironmentValues()
+            // If `name` does not include extensions, we need to try these extensions
+            var pathExtensions: Set<String> = Set(["com", "exe", "cmd", "bat"])
+            if let extensionList = currentEnvironmentValues["PATHEXT"] {
+                for var ext in extensionList.split(separator: ";") {
+                    ext.removeFirst(1)
+                    pathExtensions.insert(String(ext).lowercased())
+                }
+            }
+            // 1. The directory from which the application loaded.
+            let applicationDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: DWORD(MAX_PATH), maxSize: DWORD(MAX_PATH)
+            ) {
+                return GetModuleFileNameW(nil, $0.baseAddress, DWORD($0.count))
+            }
+            if let applicationDirectory {
+                insertExecutableAddingExtension(
+                    name,
+                    currentPath: applicationDirectory,
+                    pathExtensions: pathExtensions,
+                    storage: &possiblePaths
+                )
+            }
+            // 2. Current directory
+            let directorySize = GetCurrentDirectoryW(0, nil)
+            let currentDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: directorySize >= 0 ? directorySize : DWORD(MAX_PATH),
+                maxSize: DWORD(MAX_PATH)
```

Comment:
maxSize needs to be `DWORD(Int16.max)` to correctly handle extended length paths. `MAX_PATH` is only 260.